cpg-core/src/main/java/de/fraunhofer/aisec/cpg/frontends/cpp/CPPLanguage.kt

```diff
@@ -147,14 +150,15 @@ class CPPLanguage :
             invocationCandidates.addAll(resolveWithDefaultArgsFunc(call, scopeManager))
         }
         if (invocationCandidates.isEmpty()) {
-            /*
-             Check if the call can be resolved to a function template instantiation. If it can be resolver, we
-             resolve the call. Otherwise, there won't be an inferred template, we will do an inferred
-             FunctionDeclaration instead.
-            */
+            // Check if the call can be resolved to a function template instantiation. If it can be
+            // resolver, we resolve the call. Otherwise, there won't be an inferred template, we
+            // will do an
+            // inferred FunctionDeclaration instead.
             call.templateParametersEdges = mutableListOf()
-            if (handleTemplateFunctionCalls(null, call, false, scopeManager, currentTU)) {
-                return
+            val (ok, candidates) =
+                handleTemplateFunctionCalls(null, call, false, scopeManager, currentTU)
+            if (ok) {
+                return candidates
```

Comment:
Not quite sure how else to do it? in case the template function resolution is "ok", we need to immediately return, otherwise weird things happen.

Reply:
Ah maybe I was to unclear, the Issue is the else block after a then block with a return. The recommended way to write it is to avoid the `else {...}` and move the contained code after the If.

Reply:
Ah now I get it.